src/main/java/com/flipkart/zjsonpatch/JsonDiff.java

```diff
@@ -159,19 +171,12 @@ private static ObjectNode getJsonNode(JsonNodeFactory FACTORY, Diff diff) {
         return jsonNode;
     }
 
-    private static ArrayNode getArrayNodeRepresentation(List<Object> path) {
-        JsonNodeFactory FACTORY = JsonNodeFactory.instance;
-        ArrayNode arrayNode = FACTORY.arrayNode();
-        for (Object object : path) {
-            if (object instanceof Integer) {
-                arrayNode.add((Integer) object);
-            } else {
-                arrayNode.add(object.toString());
-            }
-        }
-        return arrayNode;
+    private static String getArrayNodeRepresentation(List<Object> path) {
+        return Joiner.on('/').appendTo(new StringBuilder().append('/'),
+                Iterables.transform(path, ENCODE_PATH_FUNCTION)).toString();
```

Comment:
If the Path contains "/", then our encoding and decoding method will fail, Since we are using "/" as delimiters.